lightning/src/onion_message/messenger.rs

```diff
@@ -365,30 +365,24 @@ impl Responder {
 	/// Creates a [`ResponseInstruction::WithoutReplyPath`] for a given response.
 	///
 	/// Use when the recipient doesn't need to send back a reply to us.
-	pub fn respond<T: OnionMessageContents>(self, response: T) -> ResponseInstruction<T> {
-		ResponseInstruction::WithoutReplyPath(OnionMessageResponse {
-			message: response,
-			reply_path: self.reply_path,
-		})
+	pub fn respond<T: OnionMessageContents>(self, response: T) -> Option<(T, ResponseInstruction)> {
+		Some((response, ResponseInstruction::WithoutReplyPath {
+			send_path: self.reply_path,
+		}))
```

Comment:
Should we drop `Option` from here and have the caller wrap if this will always be `Some`? The interface is also a little strange in that the message is passed only to be returned. Probably don't need to pass it at all.